app/controllers/orders_controller.rb

```diff
@@ -1,12 +1,15 @@
 class OrdersController < ApplicationController
   include CurrentCart
+  load_and_authorize_resource
+
   before_action :set_cart, only: [:new, :create]
   before_action :set_order, only: [:show, :edit, :update, :destroy]
 
   # GET /orders
   # GET /orders.json
   def index
-    @orders = Order.all
+    @orders = Order.all if current_user.has_role? :admin
```

Comment:
why not use it in abilities? what was the issue not being handled in abilities file?